File: java-to-kotlin-bookmanager/src/main/java/com/example/javatokotlinbookmanager/dto/user/response/UserResponse.java
package com.example.javatokotlinbookmanager.dto.user.response;

import com.example.javatokotlinbookmanager.domain.user.User;

public class UserResponse {

  private final long id;
  private final String name;
  private final Integer age;

  public UserResponse(User user) {
    this.id = user.getId();
    this.name = user.getName();
    this.age = user.getAge();
  }

  public long getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public Integer getAge() {
    return age;
  }

}

File: java-to-kotlin-bookmanager/src/main/kotlin/com/example/javatokotlinbookmanager/dto/user/response/UserResponse.kt
package com.example.javatokotlinbookmanager.dto.user.response

import com.example.javatokotlinbookmanager.domain.user.User

data class UserResponse(
    val id: Long,
    val name: String,
    val age: Int?
) {

    // 정적 팩토리 메서드 활용
    /*constructor(user: User): this(
        id = user.id!!,
        name = user.name,
        age = user.age
    )*/

    companion object {
        fun of(user: User): UserResponse {
            return UserResponse(
                id = user.id!!,
                name = user.name,
                age = user.age
            )
        }
    }
}
